### PR TITLE
price-discovery

### PR DESCRIPTION
**Pull request by dorin-iancu**

### CODE REVIEW

**Review by catalinnnn — APPROVED** [review]
Good impl

File: price-discovery/src/create_pool.rs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        let (lp_token, _, _) = contract_call.execute_on_dest_context().into_tuple();
+        self.lp_token_id().set(&lp_token.token_identifier);
+        self.total_lp_tokens_received().set(&lp_token.amount);
+
+        Ok(())
+    }
+
+    #[payable("*")]
+    #[endpoint]
+    fn accept_funds_func(&self) -> SCResult<()> {
```
Comment (sasurobert) [inline price-discovery/src/create_pool.rs:76]: make the contract SC payable and this function is not needed.

File: price-discovery/src/create_pool.rs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            EsdtTokenPayment<Self::Api>,
+            EsdtTokenPayment<Self::Api>,
+        >;
+    }
+}
+
+#[elrond_wasm::module]
+pub trait CreatePoolModule: crate::common_storage::CommonStorageModule {
+    #[endpoint(createDexLiquidityPool)]
+    fn create_dex_liquidity_pool(&self) -> SCResult<()> {
```
Comment (sasurobert) [inline price-discovery/src/create_pool.rs:27]: rename to addInitialLiquidity

File: price-discovery/src/create_pool.rs [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            token_type: EsdtTokenType::Fungible,
+            token_identifier: accepted_token_id,
+            token_nonce: 0,
+            amount: accepted_token_balance,
+        });
+
+        let dex_sc_address = self.dex_sc_address().get();
+        let contract_call = self
+            .dex_proxy(dex_sc_address)
+            .add_initial_liquidity(payments, OptionalArg::Some(ACCEPT_FUNDS_FUNC_NAME.into()));
```
Comment (sasurobert) [inline price-discovery/src/create_pool.rs:65]: optionalArg is not needed

**Review by sasurobert — APPROVED** [review]

**Pull request merged**